packages/sil-governance/src/routes/show-electorate/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { Address } from "@iov/bcp";
+import { Electorate } from "@iov/bns";
+import { Block, Hairline, Typography } from "medulas-react-components";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { RouteComponentProps } from "react-router";
+
+import { history } from "..";
+import AsideFilter from "../../components/AsideFilter";
+import BlockchainTime from "../../components/BlockchainTime";
+import ConfirmTransaction from "../../components/ConfirmTransaction";
+import Header from "../../components/Header";
+import { RootState } from "../../store/reducers";
+import { setTransactionsStateAction } from "../../store/transactions";
+import { DASHBOARD_ROUTE } from "../paths";
+import ElectorateMembers from "./components/ElectorateMembers";
+
+interface Params {
+  readonly electorateId: string;
+}
+
+const ShowElectorate = ({ match }: RouteComponentProps<Params>): JSX.Element => {
+  const dispatch = useDispatch();
+  const lastSignAndPostResult = useSelector((state: RootState) => state.transactions.lastSignAndPostResult);
+  const governor = useSelector((state: RootState) => state.extension.governor);
+  const blockchain = useSelector((state: RootState) => state.blockchain);
+
+  const address = governor ? governor.address : ("" as Address);
+  const [electorates, setElectorates] = useState<readonly Electorate[]>([]);
+  const [electorate, setElectorate] = useState<Electorate>();
+
+  useEffect(() => {
+    const updateElectorates = async (): Promise<void> => {
+      // in DOM tests, governor is not set
+      if (governor) {
+        const electorates = await governor.getElectorates();
+        setElectorates(electorates);
+      }
+    };
+    updateElectorates();
+  }, [governor]);
+
+  useEffect(() => {
+    const electorate = electorates.find(e => e.id === parseInt(match.params.electorateId, 10));
+    setElectorate(electorate);
+  }, [electorates, match.params.electorateId]);
```

Comment:
You can use `useMemo` instead of `useEffect` this will save you at least one component redraw cycle.